app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt

```diff
@@ -146,10 +150,17 @@ class NewProfileFragment : Fragment() {
                     vibSwitch = binding.vibSwitch.isChecked,
                     timeInstance = currentTime
                 )
-                profile.profileId = System.currentTimeMillis()
-                profileViewModel.insert(profile)
-                Navigation.findNavController(it)
-                    .navigate(NewProfileFragmentDirections.actionNewProfileFragmentToMainFragment())
+                if (binding.makeProfileFab.text == "Submit") {
+                    profile.profileId = System.currentTimeMillis()
+                    profileViewModel.insert(profile)
+                } else {
+                    profileListAdapter = ProfileListAdapter(profileViewModel, View(context), activity)
```

Comment:
what the point of making an instance of profileListAdapter when you don't have a recycler view in that activity? Just to access a small single line function?
Directly replace the function code if this function is called only once else make the same function in this file.